process/accountProcessor.go

```diff
@@ -132,6 +132,38 @@ func (ap *AccountProcessor) GetESDTTokenData(address string, key string) (*data.
 	return nil, ErrSendingRequest
 }
 
+// GetESDTNftTokenData returns the nft token data for a token with the given identifier and nonce
+func (ap *AccountProcessor) GetESDTNftTokenData(address string, key string, nonce uint64) (*data.GenericAPIResponse, error) {
+	observers, err := ap.getObserversForAddress(address)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		apiResponse := data.GenericAPIResponse{}
+		nonceAsString := fmt.Sprintf("%d", nonce)
+		apiPath := AddressPath + address + "/nft/" + key + "/nonce" + nonceAsString
+		respCode, err := ap.proc.CallGetRestEndPoint(observer.Address, apiPath, &apiResponse)
+		if err == nil || respCode == http.StatusBadRequest || respCode == http.StatusInternalServerError {
+			log.Info("account ESDT NFT token data error",
+				"address", address,
+				"token", key,
+				"shard ID", observer.ShardId,
+				"observer", observer.Address,
+				"http code", respCode)
+			if apiResponse.Error != "" {
+				return nil, errors.New(apiResponse.Error)
+			}
+
+			return &apiResponse, nil
+		}
+
+		log.Error("account get ESDT nft token data error", "observer", observer.Address, "address", address, "error", err.Error())
```

Comment:
is this message you will have word "error" 2 times

Reply:
fixed